x/observer/types/messages_add_block_header.go

```diff
@@ -0,0 +1,101 @@
+package types
+
+import (
+	"bytes"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/zeta-chain/zetacore/common"
+)
+
+var _ sdk.Msg = &MsgAddBlockHeader{}
+
+const (
+	TypeMsgAddBlockHeader = "add_block_header"
+)
+
+func NewMsgAddBlockHeader(creator string, chainID int64, blockHash []byte, height int64, header []byte) *MsgAddBlockHeader {
+	return &MsgAddBlockHeader{
+		Creator:     creator,
+		ChainId:     chainID,
+		BlockHash:   blockHash,
+		Height:      height,
```

Comment:
Seems the `Height` is a redundant field and used only in `ValidateBasic` for checking, but having it won't hurt at all. It's up to you.